pkg/flavor/kubernetes/nfs.go

```diff
@@ -649,7 +740,9 @@ func (flavor *Flavor) makeNFSDeployment(name string, nfsSpec *NFSSpec, nfsNamesp
 		podSpec.Spec.NodeSelector = nfsSpec.nodeSelector
 	}
 
-	if nfsSpec.podPriorityCritical {
+	// apply pod priority(supported from k8s 1.17 onwards to run critical pods in non kube-system namespace)
+	kubeletVersion, _ := flavor.getKubeletVersion()
+	if kubeletVersion != "" && semver.Compare(kubeletVersion, "v1.17.0") >= 0 {
 		podSpec.Spec.PriorityClassName = "system-cluster-critical"
 	}
```

Comment:
What happens if you have mixed environment (some nodes on < 1.17 and some on 1.17+)?  If the pod is scheduled on a 1.17+ node and that node dies, will this priority setting cause the pod to fail to be scheduled to the < 1.17 nodes?

Reply:
changed this to fetch from api-server pod.

```
time="2020-05-04T16:34:47Z" level=trace msg=">>>>> GetOrchestratorVersion" file="nfs.go:610"
time="2020-05-04T16:34:47Z" level=trace msg="obtained k8s version as v1.17.0" file="nfs.go:628"
time="2020-05-04T16:34:47Z" level=trace msg="<<<<< GetOrchestratorVersion" file="nfs.go:629"
```